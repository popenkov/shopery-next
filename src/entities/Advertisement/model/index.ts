export { type TAdvertisement } from './ad.types';
