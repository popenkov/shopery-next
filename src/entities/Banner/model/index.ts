export { type TBanner } from './banner.types';
