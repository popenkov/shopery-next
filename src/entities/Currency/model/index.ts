export * from './selectors';
export * from './types';
export * from './slices';
