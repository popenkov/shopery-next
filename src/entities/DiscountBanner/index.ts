export { type TDiscountBanner } from './model/banner';
export { DiscountBanner } from './ui';
