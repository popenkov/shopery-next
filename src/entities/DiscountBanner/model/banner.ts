export type TDiscountBanner = {
  image: string;
  title: string;
  subtitle: string;
  path: string;
};
