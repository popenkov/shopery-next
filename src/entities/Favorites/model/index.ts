export { type TWishlistProduct, type WishlistSchema } from './types';
export { selectWishlistProducts } from './selectors';
export { addToWishlist, removeProductFromWishlist, wishlistReducer } from './slices';
