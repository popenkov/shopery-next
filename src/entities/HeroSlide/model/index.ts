export { type THeroSlide } from './hero-slide.types';
