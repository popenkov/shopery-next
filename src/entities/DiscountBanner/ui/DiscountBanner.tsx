import { DetailedHTMLProps, FC, HTMLAttributes } from 'react';
import { TDiscountBanner } from '../model/banner';
import ArrowIcon from '@public/icons/icon__arrow.svg';
import cls from './DiscountBanner.module.scss';
import { AppLink } from '@/shared';
import Image from 'next/image';
import cn from 'classnames';

interface Props
    extends Omit<
            DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>,
            'title'
        >,
        TDiscountBanner {}

export const DiscountBanner: FC<Props> = ({
    image,
    title,
    subtitle,
    path,
    className,
}) => {
    return (
        <div className={cn(cls.discountBanner, className)}>
            <div className={cls.bannerImageWrapper}>
                <Image
                    className={cls.bannerImage}
                    src={image}
                    alt={title}
                    fill
                />
            </div>

            <div className={cls.bannerText}>
                <p className={cls.bannerTitle}>{title}</p>
                <p className={cls.bannerSubtitle}>{subtitle}</p>
                {/* todo */}
                {/* <a
                    className="button  button--ghost button--normal discount-banner__button icon-button"
                    href="#"
                >
                    <ArrowIcon className="icon-button__icon" />
                </a> */}
                <AppLink href={path} theme="ghost">
                    <ArrowIcon className="icon-button__icon" />
                </AppLink>
            </div>
        </div>
    );
};
