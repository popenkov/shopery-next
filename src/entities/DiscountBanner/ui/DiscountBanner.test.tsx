import React from 'react';
import { render } from '@testing-library/react';

import { TDiscountBanner } from '../model';

import { DiscountBanner } from './DiscountBanner';

const discountBannerProps: TDiscountBanner = {
  image: '/images/discount-banner.jpg',
  title: 'Discount Banner',
  subtitle: 'Get 20% off on all products',
  path: '/discounts',
};

const nextJSImageSrc = '/_next/image?url=%2Fimages%2Fdiscount-banner.jpg&w=3840&q=75';

describe('DiscountBanner component', () => {
  it('renders correctly', () => {
    const { container } = render(<DiscountBanner {...discountBannerProps} />);
    expect(container).toMatchSnapshot();
  });

  it('renders image with correct src and alt', () => {
    const { getByAltText } = render(<DiscountBanner {...discountBannerProps} />);
    const image = getByAltText(discountBannerProps.title);
    expect(image).toHaveAttribute('src', nextJSImageSrc);
  });

  it('renders title and subtitle', () => {
    const { getByText } = render(<DiscountBanner {...discountBannerProps} />);
    expect(getByText(discountBannerProps.title)).toBeInTheDocument();
    expect(getByText(discountBannerProps.subtitle)).toBeInTheDocument();
  });

  it('renders button with correct text and icon', () => {
    const { getByTestId, getByRole } = render(<DiscountBanner {...discountBannerProps} />);
    const button = getByRole('link');
    const buttonIcon = getByTestId('test-icon');

    expect(button).toHaveTextContent('Shop Now');
    expect(buttonIcon).toBeInTheDocument();
  });

  it('renders link with correct href', () => {
    const { getByRole } = render(<DiscountBanner {...discountBannerProps} />);
    const link = getByRole('link');
    expect(link).toHaveAttribute('href', discountBannerProps.path);
  });
});
