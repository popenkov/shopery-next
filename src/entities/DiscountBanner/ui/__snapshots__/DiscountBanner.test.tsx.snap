// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiscountBanner component renders correctly 1`] = `
<div>
  <div
    class="discountBanner"
  >
    <div
      class="bannerImageWrapper"
    >
      <img
        alt="Discount Banner"
        class="bannerImage"
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        sizes="100vw"
        src="/_next/image?url=%2Fimages%2Fdiscount-banner.jpg&w=3840&q=75"
        srcset="/_next/image?url=%2Fimages%2Fdiscount-banner.jpg&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fdiscount-banner.jpg&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fdiscount-banner.jpg&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fdiscount-banner.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fdiscount-banner.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fdiscount-banner.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fdiscount-banner.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fdiscount-banner.jpg&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
      />
    </div>
    <div
      class="bannerText"
    >
      <p
        class="text bannerTitle primary body_xxl left regular"
      >
        Discount Banner
      </p>
      <p
        class="text bannerSubtitle primary body_m left regular"
      >
        Get 20% off on all products
      </p>
      <a
        class="link bannerButton ghost normal"
        href="/discounts"
        tabindex="0"
      >
        <p
          class="text primary body_m left semibold"
        >
          Shop Now
        </p>
        <svg
          class="bannerButtonIcon"
          data-testid="test-icon"
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M14.2294 5.21853C13.9359 4.92627 13.461 4.92729 13.1687 5.22081C12.8765 5.51434 12.8775 5.98921 13.171 6.28147L17.935 11.025H4.75C4.33579 11.025 4 11.3608 4 11.775C4 12.1892 4.33579 12.525 4.75 12.525H17.9332L13.171 17.2676C12.8775 17.5599 12.8765 18.0347 13.1688 18.3282C13.4611 18.6217 13.9359 18.6227 14.2294 18.3304L20.258 12.3268C20.3859 12.2089 20.4728 12.0472 20.4946 11.8654C20.4983 11.8352 20.5002 11.8047 20.5002 11.774C20.5002 11.5745 20.4207 11.3833 20.2794 11.2425L14.2294 5.21853Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;
